### PR TITLE
DEV-3903-find-files

### PR DESCRIPTION
**Pull request by zamazzal**

### CODE REVIEW

**Review by nprimo — CHANGES_REQUESTED** [review]

File: subjects/devops/find-files/README.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+$find ~/
+<all files and folders in the user home>
+$find ~/ \( -type f \)
+<all files in the user home>
+$
+```
+
+May you will need to use `pipe (|)` and `&&`.
+
+> You have to use Man or Google to know more about commands flags, in order to solve this exercice!
```
[the remark this answers is not in the document]
Comment (zamazzal) [reply subjects/devops/find-files/README.md:48]: yes it should be fixed in all new exercises

File: subjects/devops/find-files/README.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+```console
+$find ~/
+<all files and folders in the user home>
+$find ~/ \( -type f \)
+<all files in the user home>
+$
+```
+
+May you will need to use `pipe (|)` and `&&`.
```
[the remark this answers is not in the document]
Comment (zamazzal) [reply subjects/devops/find-files/README.md:46]: they are different solutions to solve this exercise, and it's just a hint

File: sh/tests/find-files_test.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+
+# Unofficial Bash Strict Mode
+set -euo pipefail
+IFS='
+'
+
```
[the remark this answers is not in the document]
Comment (zamazzal) [reply sh/tests/find-files_test.sh:7]: fixed

File: subjects/devops/find-files/README.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+
+- You can use this for testing: https://assets.01-edu.org/devops-branch/find-files-example.zip
+
+- What to use : `find`
+
+- The output should be exactly like the example bellow but with the expected name
+
+```console
+$pwd
```
[the remark this answers is not in the document]
Comment (zamazzal) [reply subjects/devops/find-files/README.md:20]: I used pwd to indicate that the user should be inside the folder **`find-files-example`**

File: subjects/devops/find-files/README.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
+## find-files
+
+### Instructions
+
+"start finding ..."
+
+Create a file `find-files.sh`, which will look for and show, in the current directory and its sub-folders:
```
[the remark this answers is not in the document]
Comment (nprimo) [reply subjects/devops/find-files/README.md:7]: Ok. In any case, I feel the ask should be: 

everything that starts with an `a` **and**,
all the files ending with a `z`

Showing just one of the two would not be enough to pass this exercise

File: subjects/devops/find-files/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (nprimo) [reply subjects/devops/find-files/README.md:20]: I understand, I was showing another way to show that - this is nothing major, so it can remain as it is for me :)

File: subjects/devops/find-files/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (nprimo) [reply subjects/devops/find-files/README.md:46]: You may need to use pipes (`|`) and the logical operator `&&` 

I feel this would be clearer; up to you if you want to change it or not

**Review by nprimo — CHANGES_REQUESTED** [review]

**Comment by nprimo** [comment]
Additionally, the `README.md` should be in the `public/subject` directory if I remember right. Please, let me know if this is not the case

File: subjects/devops/find-files/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (zamazzal) [reply subjects/devops/find-files/README.md]: fixed

File: subjects/devops/find-files/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (zamazzal) [reply subjects/devops/find-files/README.md:7]: i delete the `or` in the second line but i kept the `or` in the first line you can check the script
you can check the solution file its check with or not and

File: subjects/devops/find-files/README.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+everything that starts with an `a` or,
+all the files ending with a `z` or,
```
Comment (nprimo) [inline subjects/devops/find-files/README.md]: There is an extra `or` at the end of this line. I would finish the line with a `.`

File: subjects/devops/find-files/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (nprimo) [reply subjects/devops/find-files/README.md:7]: For clarity, I would rephrase it as follow

```
everything that starts with an `a` or files ending with a `z`
```

In any case, the example below clarify what is supposed to have as a result of the script

**Review by nprimo — APPROVED** [review]
LGTM! If you want to keep the current version, feel free to resolve the conversation where I suggest a clearer (very subjective) option

File: subjects/devops/find-files/README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (zamazzal) [reply subjects/devops/find-files/README.md:7]: done, i explained cause the students will join the branch with no experience in linux so this will be hard for them to know what the different between all files and everything in linux
thank you
